### PR TITLE
FEAT: Async Shell Script Execution whenever Sketchware Starts

### PR DESCRIPTION
**Pull request by 1tsMeMario**
Run a custom shell script whenever you start sketchware. Can be used for restoration or backup purposes or if you are advanced in shell scripting then whatever you can think of.

### CODE REVIEW

**Comment by onyxfluxx** [comment]
Setting is in Mod Settings, by default it is turned off. Root and non-root users can use this script. Must manually create the script in root of Internal Storage. Name file sketchwarescript.sh

File: app/src/main/java/com/besome/sketch/MainActivity.java
Comment (JavkhlanK) [inline app/src/main/java/com/besome/sketch/MainActivity.java:177]: Would a file at the root of Internal storage really be a good idea? I'd say put it into /Internal storage/.sketchware/ or /Internal storage/sketchware/ at least.

File: app/src/main/java/com/besome/sketch/MainActivity.java
Comment (JavkhlanK) [inline app/src/main/java/com/besome/sketch/MainActivity.java:179]: Maybe make it show a toast or at least an entry in debug.txt, no average user's gonna be looking in logcat for script errors.

File: app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java
Comment (JavkhlanK) [inline app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java:49]: Why? This will only break compatibility with older Sketchware Pro versions.

**Review by JavkhlanK — CHANGES_REQUESTED** [review]
There's a few things that I think should be addressed, though I don't see this feature be useful for many people. Don't automatization apps already provide app launches as triggers? For example, Automate already has such a thing.

File: app/src/main/java/com/besome/sketch/MainActivity.java
Comment (onyxfluxx) [reply app/src/main/java/com/besome/sketch/MainActivity.java:177]: this was actually where i had it in my last uodate but i had to do a revert so yes ill update that

File: app/src/main/java/com/besome/sketch/MainActivity.java
Comment (onyxfluxx) [reply app/src/main/java/com/besome/sketch/MainActivity.java:179]: i tried doing a toast but it never showed up after i made this asynchronous. maybe you know a solution?

File: app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java
Comment (onyxfluxx) [reply app/src/main/java/mod/hilal/saif/activities/tools/ConfigActivity.java:49]: the pallete part ill change it back lol i just thought it was a spelling error sorry

File: app/src/main/java/com/besome/sketch/MainActivity.java
Comment (JavkhlanK) [reply app/src/main/java/com/besome/sketch/MainActivity.java:179]: Sounds like a thread issue. UI things like toasts can only happen on the main/UI thread, and since AsyncTasks run in a new thread, try using `runOnUiThread(Runnable)` to display a toast.

**Comment by ilyassesalama** [comment]
It's been almost a year now. @1tsMeMario will you make any changes to this pull request anytime soon?

**Comment by JavkhlanK** [comment]
> It's been almost a year now. @ 1tsMeMario will you make any changes to this pull request anytime soon?

> though I don't see this feature be useful for many people. Don't automatization apps already provide app launches as triggers? For example, Automate already has such a thing.

**Comment by 1tsMeMario** [comment]
yall never wanted to approve it, thats on yall. i gave up on it on the
public end and just improved it on my end only.

**Comment by ilyassesalama** [comment]
@1tsMeMario As long as this pull request is active, we're still waiting for your changes. We love people who contribute to making Sketchware Pro even better. However, any contribution must be thoroughly reviewed before merging it to make sure it won't affect our users. You know, they have projects that must stay safe and we're responsible for that.

The last thing you said is "the pallete part ill change it back lol i just thought it was a spelling error sorry". So, we're waiting for your change to merge this pull request. You didn't even get back to us about this pull request. We would have clarified what is wrong/should be done.

Again, we're waiting for your changes, and why not some new awesome contributions in the future? Thanks for the time you've spent contributing to Sketchware Pro.

**Pull request closed without merging**
